math/src/main/java/de/florianmichael/rclasses/math/Arithmetics.java

```diff
@@ -265,4 +265,51 @@ public static boolean isDecimal(final double value) {
         return value % 1 != 0;
     }
 
+    /**
+     * @param u The first value
+     * @param v The second value
+     * @return The greatest common divisor of u and v
+     */
+    public static long gcd(long u, long v) {
+        // Java re-write of the Binary GCD algorithm rust implementation on https://en.wikipedia.org/wiki/Binary_GCD_algorithm
+
+        if (u < 0 || v < 0) {
+            throw new UnsupportedOperationException("Both values must be positive");
+        }
+
+        // A rule is that if u == 0 then v is the greatest common divisor and vice versa
+        if (u == 0)
+            return v;
+        else if (v == 0)
+            return u;
+
+        // gcd(pow(2, uTrailingZeroes) u, pow(v, vTrailingZeroes v) = pow(2, zeroes)  gcd(u, v) with u, v odd and k = min(i, j)
+        // pow(2. zeroes) is the greatest power of two that divides both pow(2, uTrailingZeroes) u and pow(2, vTrailingZeroes) v
+        int uTrailingZeroes = Long.numberOfTrailingZeros(u);
+        u >>= uTrailingZeroes;
+        int vTrailingZeroes = Long.numberOfTrailingZeros(v);
+        v >>= vTrailingZeroes;
+        long zeroes = Math.min(uTrailingZeroes, vTrailingZeroes);
+
+        while (true) {
+            // Swap if necessary so u <= v
```

Comment:
Should check if both u and v are odd at the beginning, maybe just throw an exception then.